include/deal.II/lac/la_parallel_block_vector.templates.h

```diff
@@ -779,6 +779,56 @@ namespace LinearAlgebra
               MemoryConsumption::memory_consumption (this->components));
     }
 
+
+
+    template <typename Number>
+    template <typename FullMatrixType>
+    void
+    BlockVector<Number>::multivector_inner_product(FullMatrixType &matrix,
+                                                   const BlockVector<Number> &V,
+                                                   const bool symmetric) const
+    {
+      const unsigned int m = this->n_blocks();
+      const unsigned int n = V.n_blocks();
```

Comment:
Shouldn't you assert that `n==m`?

Reply:
nope, only in case it's symmetric which happens below. Otherwise it's perfectly fine to have non-square matrices.